core/context.go

```diff
@@ -286,14 +301,27 @@ func (ctx *Context) UpdateBackend(vsID, rsID string, weight int32) (int32, error
 	// Save the old backend weight and update the current backend weight.
 	result, rs.options.Weight = rs.options.Weight, weight
 
+	// Currently the backend options are changing only the weight.
```

Comment:
Do you want to modify the Weight values to be saved as Weight (first set value) and weight (current value) as BackendOptions.Host and BackendOptions.hostip?
And what about modifying it to sync for all changes?